mdsuite/calculators/green_kubo_self_diffusion_coefficients.py

```diff
@@ -196,11 +195,15 @@ def ensemble_operation(self, ensemble):
             / self.experiment.units.time**2
             * tfp.stats.auto_correlation(ensemble, normalize=False, axis=1, center=False)
         )
+        self.count += vacf.shape[0]
 
         # average particles, sum dimensions
-        vacf = tf.reduce_sum(tf.reduce_mean(vacf, axis=0), -1)
-        self.sigmas.append(cumtrapz(vacf, x=self.time))
-        self.vacfs.append(vacf)
+        return_vacf = tf.reduce_sum(tf.reduce_sum(vacf, axis=0), -1)
+        self.sigmas.append(
+            cumtrapz(tf.reduce_sum(tf.reduce_mean(vacf, axis=0), -1), x=self.time)
```

Comment:
I don't understand, why you duplicated the code here? Also consider using `tf.reduce_sum(vacf, axis=(0, -1))` for performance reasons.

Reply:
One provides an uncertainty and one provides the actual acf which is normalized later to avoid false normalization when we have mini-batched or remainder ensembles. I could remove the sigmas line but then we would have no uncertainty on the prediction so thus far this is the best way.